### PR TITLE
[Settings] Clean up naming and typings

### PR DESCRIPTION
**Pull request by psealock**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR:
* Fixes duplicate naming for `pageSettings` v. `settingsPage` which is of type `SettingsPage`. 
* The types in the `global.d.ts` file were updated so IDE's can read them
* Update naming to reflect types. Instead of `pages: type of window.wcSettings.admin.settingsPages` its now `settingsData: SettingsData`

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Build works, tests pass
2. Enable settings feature flag and activate gutenberg
3. Settings should work, reflecting the route, and load without error

image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

Updates unreleased feature

</details>

### CODE REVIEW

File: packages/js/settings-editor/changelog/52806-fix-settings-typings [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: tweak
+Comment: Updates unreleased feature
```
Comment (ilyasfoo) [inline packages/js/settings-editor/changelog/52806-fix-settings-typings:3]: Nit: I think we need a bit more specific with the changelog here, we've been called out once for having an overly generic description 😅

File: plugins/woocommerce/changelog/52806-fix-settings-typings [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: tweak
+Comment: Updates unreleased feature
```
Comment (ilyasfoo) [inline plugins/woocommerce/changelog/52806-fix-settings-typings:3]: Same as above comment

**Review by ilyasfoo — APPROVED** [review]
Tests well, no build and UI errors, and code looks good! I have a nit comment on changelog but pre-approving

**Comment by psealock** [comment]
>Nit: I think we need a bit more specific with the changelog here, we've been called out once for having an overly generic description 😅

Thanks @ilyasfoo, I updated the changelogs

**Pull request merged**